### ISSUE TITLE
Check Types for SocketOptions elements in NetworkSocket object

### ISSUE THREAD

**Issue by gtback** [CybOXProject/schemas#292]
While adding tests to python-cybox, I encountered some unexpected types on some of the socket options. It may just be that some of the options are different depending on the context, but it might be worth an exhaustive review to clarify which options are valid in which contexts (and add this to the documentation). At a minimum:
- Java
- Linux/Unix C programming
- Windows API

**Comment by ikiril01** [comment]
@gtback just to clarify what you mean by types, are you saying that some of the datatypes on the elements in the SocketOptionsType might be incorrect?

**Comment by gtback** [comment]
Yes, I was referring to the `type` of each `element` (e.g. boolean, StringObjectPropertyType, UnsignedIntegerPropertyType). I don't remember specifically which elements caught my attention. 

It also seems that most of the options assume the socket is on a POSIX operating system (i.e. an option to `setsockopt`). There also seem to be some differences in the options supported by different environments/languages. I haven't examined the documentation closely, though they do appear to be pretty similar... more similar than I thought they were when originally making this issue.

* [Java](https://docs.oracle.com/javase/8/docs/api/java/net/Socket.html)
* [POSIX](http://linux.die.net/man/3/setsockopt)
* [Windows (Winsock)](https://msdn.microsoft.com/en-us/library/windows/desktop/ms740476%28v=vs.85%29.aspx)
